### ISSUE TITLE
[BUG] clusterUI templated/batch recipe py3 issues (/regression re: inputsByTask?)

### ISSUE THREAD

**Issue by barentine** [python-microscopy/python-microscopy#487]
**Describe the bug** 
Several regressions or py3 compatibility issues are currently present in clusterUI and hold up using it to batch recipes. Not included here is an enhancement proposal to allow templating more than just files / GUI editing, as the write-once cluster philosophy otherwise implies needing to write a new recipe for e.g. each coverslip you image #473 .

- [x] There is a bytes/str issue in creating folders
![image](https://user-images.githubusercontent.com/31105780/95333500-dce59d80-087a-11eb-986e-663e0dc16b71.png) 
```
Traceback: File "/home/smeagol/miniconda3/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner 41. response = get_response(request) File "/home/smeagol/miniconda3/lib/python3.6/site-packages/django/core/handlers/base.py" in _legacy_get_response 249. response = self._get_response(request) File "/home/smeagol/miniconda3/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response 187. response = self.process_exception_by_middleware(e, request) File "/home/smeagol/miniconda3/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response 185. response = wrapped_callback(request, *callback_args, **callback_kwargs) File "/home/smeagol/code/python-microscopy/PYME/cluster/clusterUI/clusterbrowser/views.py" in mkdir 141. clusterIO.put_file(newDirectory, '') File "/home/smeagol/code/python-microscopy/PYME/IO/clusterIO.py" in put_file 960. raise TypeError('data should be bytes (not a unicode string)') Exception Type: TypeError at /files/mkdir/CALIBRATION/ Exception Value: data should be bytes (not a unicode string)
```
- [x] There is a py3 compatibility issue between str / bytes in the templated recipe `view`, as clusterIO read returns bytes. #476. 
- [x] fixing that by decoding puts a multiline str (the recipe in yaml format) in json, which breaks the templated view since we json dumps/loads
```
  File "/home/ubuntu/miniconda3/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/code/python-microscopy/PYME/cluster/rulenodeserver.py", line 344, in _poll_tasks
    self._update_tasks()
  File "/home/ubuntu/code/python-microscopy/PYME/cluster/rulenodeserver.py", line 215, in _update_tasks
    taskID, cost = rater.get_local()
  File "/home/ubuntu/code/python-microscopy/PYME/cluster/rulenodeserver.py", line 113, in get_local
    taskID, cost = self.next()
  File "/home/ubuntu/code/python-microscopy/PYME/cluster/rulenodeserver.py", line 107, in next
    return self.__next__()
  File "/home/ubuntu/code/python-microscopy/PYME/cluster/rulenodeserver.py", line 78, in __next__
    task = json.loads(filled_template)
ValueError: Unexpected character in found when decoding object value
```
Which can be alleviated with the `strict=False` kwarg on `json.loads`, however that unveils another issue, breaking at the same place, due to `inputsByTask` being double dumps'd, at least with the templated recipe `view`. In any event, `rulenodeserver.py` itself can be made simpler/cleaner/more robust, as in #485. I haven't tested that fix with the `HTTPRulePusher.RecipePusher` because by that point I'd switched over the using the `_rules.RecipeRule`, some version of which I am/we are more invested in using going forward. Obviously I tried to keep these issues in separate PRs, but ran into them one after the other trying to do one thing so am posting here together.

**To Reproduce**
1. run a cluster on py3
2. run clusterUI
3. batch a templated recipe over multiple files

**Expected behavior**
post/assign/complete the recipe

**Platform (please complete the following information - or copy-paste from error dialog if shown):**
 - OS: [e.g. Win10] all ubuntu, ranging from 14.04 to 18.04
 - Python version: 3.6 / 3.7
 - numpy version: 1.14 to 1.19
 - wxpython version [if a UI bug]:

**Comment by barentine** [comment]
As far as priorities, I tried to keep things separated out. @David-Baddeley let's slack about times to meet re: #486 
 `RecipeRule` vs refactoring it vs `HTTPRulePusher.RecipePusher`, but tomorrow would be great.

Meanwhile, if merging #476 and #485 are acceptable there's a chance they fix my issue and let me batch all the recipes I need to get back to my collaborators (bandwidth of the given recipe on a single machine has been pretty impractical, but will either be hacking that through tomorrow or imaging)

**Comment by barentine** [comment]
addressed with #476, #541, #561, and #572
